compiler/luci/service/src/Nodes/CircleAdd.test.cpp

```diff
@@ -56,6 +56,235 @@ TEST(ShapeRuleTest, different_input_shapes_add)
   ASSERT_EQ(5, shape.dim(2).value());
 }
 
+TEST(ShapeRuleTest, add_dynamic_shape_non_1)
+{
+  luci::CircleInput input_1;
+  luci::CircleInput input_2;
+  luci::CircleAdd add;
+
+  input_1.shape({1, 4, 3, 1});
+  input_1.shape_status(luci::ShapeStatus::VALID);
+
+  input_2.shape({1, 1, 1});
+  input_2.shape_status(luci::ShapeStatus::VALID);
+  input_2.dim(0).unset();
```

Comment:
Q) can we SI for `input_2` having not `1` like `{?, 4, 3}` or `{?, 3, 1}` ?

Reply:
>  what SI is?

ah... shape inference

Reply:
Current test checks 4D vs 3D (w/ all 1's); I understand broadcast here is `rank up` + `expand dims`.
I was curious how it would work with having not 1 like `1, 3, 1`.
Anyway current code looks OK.

Reply:
> I was curious how it would work with having not 1 like 1, 3, 1.

{?, 3, 1} will be first broadcasted to {1, ?, 3, 1}, and the output shape will be {1, 4, 3, 1}.